### ISSUE TITLE
Broken robots and weird spawning!

### ISSUE THREAD

**Issue by Arkham4ever** [kyranf/robotarmyfactorio#134]
Hey guys! (please excuse my terrible english! :-P)

first of all, this is a very great mod and i was looking for something like this quite a while because i think thats whats factorio missing the most! it really adds a ton of fun to the game!

well, i installed the mod just yesterday, working fine but it took me a bit to figure things out. today i wanted to add the "better roboports" mod to the game. as the name intends, this mod adds some robot hangars with quite a big reach at the cost of a ton of electricity

there are some droid guard stations in my base and since i loaded the game after installing the better roboports mod, the droids have problems with spawning. regardless of what bot i put into the guard station, only one will be spawned and it remains still until a creep gets near it or have me manually remove the droid out of this exact spot.

however, if there are biters coming, the droids start acting normal again until the next one is spawned. already spawned droids behaive normal but the guard station cant seem to spawn moving droids anymore..

is this me just being dumb at one point or did i really mess up this wonderfull mod and are forced to start over?

**Comment by kyranf** [comment]
have you adjusted the squad sizes with the droid settings module combinator?

**Comment by Arkham4ever** [comment]
no.. to be honest, im totally dumb when it comes to this whole combinator thing.. could you explain to me how i do that?

**Comment by kyranf** [comment]
Okay, that's good to know.

The guard station has spawning logic which counts in a radius of 32 tiles around it, checking for total droid count in that area. If  the droid count exceeds the guard station squad size (not the hunt size! That's unrelated) it will stop spawning. 

You say they spawn on top of eachother? Are they colliding inside each other? Or spaced out like what should be happening with the Droid Assemblers?

**Comment by Arkham4ever** [comment]
well its both.. sometimes the droids simply spawn on top of each other and sometimes theyre spawning in different spots but wont move at all unless biters get too close to them.. i also noticed a pretty small group of droids roaming through the base, sometimes they get super eager to move to a specific location for whatever reason..

i tried disabling the mod temporarily, and messed around with the creative mode.. it seems that theres a special condition that has to be met in order to activate this behaviour..

**Comment by kyranf** [comment]
Are you maybe expecting to have squads hunting? or do you want them to wait around and defend? 

If you WANT them to move out and attack biters, you are using the wrong building. the Guard station only spawns droids that wait around until biters come within visual distance. The "Droid Assembler" building is the one you need for making squads go out and attack and they will spawn properly etc.

Is that what is happening here?

**Comment by Arkham4ever** [comment]
nope, i just want them to move around the guard station building but they keep wandering off nonetheless..

is it maybe because i put the guard station too close together?

**Comment by kyranf** [comment]
Can you take a screenshot of your setup for me?

**Comment by Arkham4ever** [comment]
sure! in the meanwhile i deinstalled the mod since it caused so much "unwanted" moving in and around my base, give me a bit to re do what i did :-)

**Comment by kyranf** [comment]
Thank you.. I completely understand, sorry for the negative experience you have had with my mod
